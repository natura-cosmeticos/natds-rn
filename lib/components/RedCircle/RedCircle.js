import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';

import { ThemeNaturaLight } from '../../themes/ThemeNaturaLight';

export const RedCircle = ({ whereToRender }) => (
    <View
      style={{
        backgroundColor: ThemeNaturaLight.palette.error.main,
        borderRadius: 50,
        height: 10,
        position: 'absolute',
        right: whereToRender === 'AppBar' ? 17 : 0,
        top: whereToRender === 'AppBar' ? 17 : 13,
        width: 10,
      }}
    />
);

RedCircle.propTypes = {
  whereToRender: PropTypes.string,
};

RedCircle.defaultProps = {
  whereToRender: '',
};
