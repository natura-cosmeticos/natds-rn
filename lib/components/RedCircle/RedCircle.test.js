import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';

import { RedCircle } from './RedCircle';

describe('RedCircle component default', () => {
  it('Should render RedCircle RedCircle ', () => {
    const circle = renderer.create(<RedCircle />).toJSON();

    expect(circle).toMatchSnapshot('RedCircle snapshot');
  });
});
