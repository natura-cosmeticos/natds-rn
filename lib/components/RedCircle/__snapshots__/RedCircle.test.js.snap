// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedCircle component default Should render RedCircle RedCircle : RedCircle snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E74627",
      "borderRadius": 50,
      "height": 10,
      "position": "absolute",
      "right": 0,
      "top": 13,
      "width": 10,
    }
  }
/>
`;
